### PR TITLE
Add support in Patterns::Tools::Convert for std::array

### PR DESCRIPTION
**Pull request by luca-heltai**
I have tried with some template magic to get this working within the rest of the list_types, but did not manage. With cxx17 support this could be achieved without duplicating code, but for the moment this fix will do.

### CODE REVIEW

**Comment by luca-heltai** [comment]
Fixes #13218.

**Review by tamiko — APPROVED** [review]
OK after fixing the indentation of the test source file :-)

File: include/deal.II/base/patterns.h [hunk 1]
```diff
@@ -1909,6 +1914,74 @@ namespace Patterns
...
+        std::transform(
+          t.cbegin(), t.cend(), vec.begin(), [&base_p](const auto &entry) {
+            return Convert<typename T::value_type>::to_string(entry, *base_p);
+          });
+
+        std::string s;
+        if (vec.size() > 0)
+          s = vec[0];
+        for (unsigned int i = 1; i < vec.size(); ++i)
+          s += p->get_separator() + " " + vec[i];
```
Comment (bangerth) [inline include/deal.II/base/patterns.h:1957]:
```suggestion
          s += p->get_separator() + ' ' + vec[i];
```

**Pull request merged**